### DISCUSSION TITLE
How to list dependencies?

### DISCUSSION THREAD

**Discussion by mtelka** [python-poetry/poetry#8669] [Q&A] [answered]
I need to list (print) dependencies for a project.  Let say I do have a project (`doff_cover` in this case) with following in its `pyproject.toml`:
```
[tool.poetry.dev-dependencies]
pytest-cov = "^4.1.0"
pytest-datadir = "^1.5.0"
pytest-mock = "^3.12.0"
pycodestyle = ">=2.9.1"
flake8 = "^5.0.4"
pyflakes = "^2.5.0"
pylint = "^2.17.7"
pylint-pytest = "^1.1.3"
pydocstyle = "^6.1.1"
black = "^23.10.1"
isort = "^5.11.5"
doc8 = "1.1.1"
```
Is there a way how to list these dependencies?  For example to get `pip freeze`-compatible output?

Yes, I can install those dependencies using `poetry install`, but I do not want to install them, I just need to print them.

Thank you.

PS: There is nice documentation about [poetry dependencies](https://python-poetry.org/docs/managing-dependencies/), but I failed to find there the information I'm looking for.

**Comment by dimbleby** [comment]
`poetry list` lists the available commands.

**Comment by squarism** [comment]
`poetry run pip list` runs `pip list` inside of the project context (ie: the lock file). This is great because it's project isolated, ideally content computed off of hashes and repeatable all things being equal.

I have recently learned this does not consider on-disk state so you have to be careful.  Probably run `poetry install` before you do your print out.

**Comment by gmarkingembed** [comment]
`poetry show` lists all packages in the environment. You can also pass a package into the command `poetry show <package>` to get a detailed summary including its dependencies and required by packages in the env.

**Reply by squarism** [reply]
`poetry show` doesn't show the versions as a list.  You can see the versions with `poetry show --tree` but then it's n  oisy because of transitive dependencies.  You can use `poetry show --top-level` but this does not show versions either   and cannot be used with `--tree`.  It's for this reason and getting people used to `poetry` coming from `pip` that I   just demonstrate/share `poetry run pip list`.

Maybe there are other ways too.

**Reply by interifter** [reply]
An update to this, for anyone else ending up here. `poetry show --top-level` shows versions now:

`Poetry (version 2.0.1)`

```
> poetry show --top-level
chardet            5.2.0             Universal encoding detector for Python 3
click              8.1.8             Composable command line interface toolkit
...
```

**Comment by mtelka** [comment, marked as answer]
The best solution that fits my needs is https://pypi.org/project/pyproject-installer/
